### ISSUE TITLE
SA1009 must not be triggered when the closing parenthesis followed by '>' operator 

### ISSUE THREAD

**Issue by RomanDidenko** [DotNetAnalyzers/StyleCopAnalyzers#2409]
Next code reports SA1009 warning but should not:

```cs
collection.Where(g => g.Count() > 1);
```

**Comment by sharwell** [comment]
Yep, accidentally introduced this when fixing #2267.

**Comment by RomanDidenko** [comment]
Added tests to cover failure: #2410
